integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -3429,6 +3429,29 @@ fn test_limit_contract_functions_number_upgrade() {
     ));
 }
 
+#[test]
+/// Test that if a node's shard assignment will not change in the next epoch, the node
+/// does not need to catch up.
+fn test_catchup_no_sharding_change() {
+    init_integration_logger();
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap()], 1);
+    genesis.config.epoch_length = 5;
+    let chain_genesis = ChainGenesis::from(&genesis);
+    let mut env = TestEnv::builder(chain_genesis)
+        .clients_count(1)
+        .validator_seats(1)
+        .runtime_adapters(create_nightshade_runtimes(&genesis, 1))
+        .build();
+    // run the chain to a few epochs and make sure no catch up is triggered and the chain still
+    // functions
+    for h in 1..20 {
+        let block = env.clients[0].produce_block(h).unwrap();
+        let (_, res) = env.clients[0].process_block(block.unwrap().into(), Provenance::PRODUCED);
+        res.unwrap();
+        assert_eq!(env.clients[0].chain.store().iterate_state_sync_infos(), vec![]);
+    }
+}
```

Comment:
Could we assert that blocks to catch up (`ColBlocksToCatchUp`) is also always empty?